src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -110,6 +116,9 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
      * Fills up all the placeholders of this window.
      */
     void fillInnerParts() {
+        greetingBar = new GreetingBar(logic.getFilteredPersonList());
+        greetingBarPlaceholder.getChildren().add(greetingBar.getRoot());
+
         personListPanel = new PersonListPanel(logic.getFilteredPersonList());
         personListPanelPlaceholder.getChildren().add(personListPanel.getRoot());
```

Comment:
Similar to above, I believe a better way of writing this logic is to expose a `update_children`-like function!

Reply:
I can rewrite this later if you are unsure how to